### PR TITLE
Added disclaimer for products in beta.

### PR DESCRIPTION
**Pull request by spsiddarthan**
#### Proposed Changes

* The soon-to-be-launched Jetpack Social Advanced plan is in Beta, we are pricing it at 180$ for the first year, but we are going to double the price after one year. We need to add this to our terms and conditions.

#### Testing Instructions

* Do `yarn start` to boot up your local calypso after checking out to this branch. 
* Go to `http://calypso.localhost:3000/checkout/jetpack/jetpack_social_advanced_yearly` and look at the newly added disclaimer. 
* 
Screenshot 2023-02-02 at 15 05 58

* Go to   `http://calypso.localhost:3000/checkout/jetpack/jetpack_social_basic_yearly` and make sure that the newly added disclaimer is gone since it's not applicable to the selected product. 

#### Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Related to #

### CODE REVIEW

File: client/my-sites/checkout/composite-checkout/components/jetpack-product-pricing-disclaimer.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	const product_names = products.map( ( product ) => product.product_name );
+
+	const showJetpackProductPricingDisclaimer = JETPACK_FIRST_YEAR_50_PERCENT_OFF_PRODUCTS.filter(
+		( slug ) => product_slugs.includes( slug )
+	);
+	if ( showJetpackProductPricingDisclaimer.length > 0 ) {
+		return (
+			<CheckoutTermsItem key="jetpack-product-pricing-disclaimer">
+				{ translate(
+					'The following products are currently in development: %(productNames)s Enjoy using them at half price over the next year while we continue to develop the features.',
```
Comment (sirbrillig) [inline client/my-sites/checkout/composite-checkout/components/jetpack-product-pricing-disclaimer.tsx]: Do you want a period between the list of product names and the next sentence?
```suggestion
					'The following products are currently in development: %(productNames)s. Enjoy using them at half price over the next year while we continue to develop the features.',
```

**Review by sirbrillig — COMMENTED** [review]
I leave final approval to your team, but from a Payments perspective this looks good to me! I made one suggestion to the text. It looks ok in the screenshot but that's just because the text happened to wrap at an elegant place.

File: client/my-sites/checkout/composite-checkout/components/jetpack-product-pricing-disclaimer.tsx (hunk 1 above)
Comment (spsiddarthan) [reply client/my-sites/checkout/composite-checkout/components/jetpack-product-pricing-disclaimer.tsx]: We are doing this only for Jetpack Social Advanced for now -  p1675291379826429/1675244371.316719-slack-C02JJ910CNL, so I have updated the text and the code to reflect that.

**Review by gmjuhasz — APPROVED** [review]
This looks good to me!

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7798871

Thank you @spsiddarthan for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7798871).

**Pull request merged**